equinox/nn/_embedding.py

```diff
@@ -217,24 +223,29 @@ def __call__(
             )
 
         with jax.ensure_compile_time_eval():
-            if embedding_size in internal_rope_embedding_cache:
-                freqs_cis = internal_rope_embedding_cache[embedding_size]
-                freqs_cis_seq_len, _ = freqs_cis.shape
+            if (embedding_size, self.dtype) in internal_rope_embedding_cache:
+                freqs_pair = internal_rope_embedding_cache[(embedding_size, self.dtype)]
+                freqs_cis_seq_len, _ = freqs_pair[0].shape
```

Comment:
NitL this could be done more neatly as `freqs_cos, freqs_sin = ...`. This would also save you needing to index a few lines lower down.

(This is one of my common bugbears, to prefer unpacking over indexing -- much harder for bugs to slip through this way.)